src/sysinfo/nvapi_adapter_registry.cpp

```diff
@@ -40,6 +40,11 @@ namespace dxvk {
                 delete nvapiAdapter;
         }
 
+        m_dxgiFactory = dxgiFactory;
```

Comment:
Might as well replace `dxgiFactory` with `m_dxgiFactory` everywhere here. When `Initialize` fails (`return false`), the current instance of `NvapiAdapterRegistry` will be deleted anyway (happens in `NvAPI_Initialize`).